jetty-client/src/main/java/org/eclipse/jetty/client/HttpRequest.java

```diff
@@ -353,7 +381,7 @@ public Request attribute(String name, Object value)
     }
 
     @Override
-    public HttpFields getHeaders()
+    public HttpFields.Mutable getHeaders()
```

Comment:
Actually now that I have added the `put(HttpField)`, `add(HttpField)` etc. methods it could be an immutable return.
So currently this supports both old style (get the fields as a builder and modify) and the new style (use `add`,`put`, `set`).

We can keep it like this, go back to just old style (removing new methods) or change to immutable to just support new style.  Your thoughts?

Reply:
See #4808. Let's finish this and then discuss - currently I'm leaning for `Request headers(Consumer<HttpFields.Mutable>)`

Reply:
I did try the consumer approach and it worked well enough, but was a little bit more verbose.   Will defer this to #4808